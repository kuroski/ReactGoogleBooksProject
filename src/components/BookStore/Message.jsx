import styled from 'styled-components'

const Message = styled.div`
    font-family: Roboto;
    text-align: center;
    background-color: #212121;
    padding: .5rem;
    color: white;

    &:empty {
        display: none;
    }
`

export default Message