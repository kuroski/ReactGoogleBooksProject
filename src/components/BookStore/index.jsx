import React, {Component} from 'react'
import SearchForm from '../SearchForm'
import BookShelf from '../BookShelf'
import Pagination from '../Pagination'
import * as booksApi from '../../api'
import {FormattedMessage} from 'react-intl'

class BookStore extends Component {
  constructor(props) {
    super(props)
    this.state = {
      books: [],
      message: '',
      currentSearchTerm: '',
      currentPage: 0,
      pageCount: 0
    }

    this.executeBookSearch = this.executeBookSearch.bind(this)
    this.executePageChange = this.executePageChange.bind(this)
    this.searchAllBooks = this.searchAllBooks.bind(this)
    this.executeFavoriteBook = this.executeFavoriteBook.bind(this)
  }

  executeBookSearch(term) {
    if(!term) {
      this.setState({
        message: <FormattedMessage id="app.errors.nosearchtermprovided" defaultMessage="Nenhum termo de pesquisa foi digitado"></FormattedMessage>
      })
      return
    }

    return this.searchAllBooks(term)
  }

  executePageChange(toPage) {
    return this.searchAllBooks(this.state.currentSearchTerm, toPage)
  }

  searchAllBooks(term, page = 0) {
    return booksApi.all(term, page)
      .then(books => {
        this.setState({
          message: '',
          books: books.items,
          currentSearchTerm: term,
          currentPage: page,
          pageCount: Math.ceil(books.totalItems / 10)
        })
      })
  }

  executeFavoriteBook(bookId) {}

  render() {
    return (
      <div className="BookStore">
        <div>{this.state.message}</div>
        <SearchForm onSubmit={this.executeBookSearch} />
        <BookShelf onFavorite={this.executeFavoriteBook} books={this.state.books} />
        <Pagination currentPage={this.state.currentPage} pageCount={this.state.pageCount} onPageChange={this.executePageChange} />
      </div>
    )
  }
}

export default BookStore
