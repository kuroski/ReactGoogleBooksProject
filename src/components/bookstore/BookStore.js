import React, { Component } from 'react';

class BookStore extends Component {
  render() {
    return (
      <div className="BookStore">
        Hello Bookstore
      </div>
    );
  }
}

export default BookStore;
