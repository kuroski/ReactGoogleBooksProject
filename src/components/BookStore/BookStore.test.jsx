import React from 'react'
import BookStore from './'
import renderer from 'react-test-renderer'

describe('BookStore', () => {
  it('renders correctly', () => {
    const BookStoreTree = renderer.create(
      <BookStore />
    ).toJSON()
    expect(BookStoreTree).toMatchSnapshot()
  })
})
