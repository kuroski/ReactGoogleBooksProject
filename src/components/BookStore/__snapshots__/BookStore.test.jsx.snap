exports[`BookStore renders correctly 1`] = `
<div
  className="BookStore">
  Hello Bookstore
</div>
`;
